api/v1alpha1/backendsecurity_policy.go

```diff
@@ -71,6 +73,11 @@ type BackendSecurityPolicySpec struct {
 	// +optional
 	AWSCredentials *BackendSecurityPolicyAWSCredentials `json:"awsCredentials,omitempty"`
 
+	// AzureAPIKey is a mechanism to access Azure OpenAI backend(s). The API key will be injected into the api-key header.
+	//
+	// +optional
+	AzureAPIKey *BackendSecurityPolicyAzureAPIKey `json:"azureAPIKey,omitempty"`
```

Comment:
what's your take on instead of having this, adding a configuration knob in the existing `BackendSecurityPolicyAPIKey` to specify the header key that defaults to "authorization" ? I think "bearer: " portion makes it tricky though...

Reply:
not a fan tbh. I'd rather switch to ad hoc mutation than clutter unrelated auth approaches. the thing is the other one is used for oauth and this has nothing to do with oauth.

Reply:
oh i meant here https://github.com/envoyproxy/ai-gateway/blob/bb2a476eeb7fe14495a64ced1396c65a07ea2cd3/api/v1alpha1/backendsecurity_policy.go#L94 but clunky anyways i agree